frontend/src/scenes/pipeline/PipelineAppConfiguration.tsx

```diff
@@ -0,0 +1,122 @@
+import { LemonSkeleton, LemonWidget, Tooltip } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { Form } from 'kea-forms'
+import { Field } from 'lib/forms/Field'
+import { IconLock } from 'lib/lemon-ui/icons'
+import { LemonMarkdown } from 'lib/lemon-ui/LemonMarkdown'
+import React, { useEffect, useState } from 'react'
+import {
+    defaultConfigForPlugin,
+    determineInvisibleFields,
+    determineRequiredFields,
+    getConfigSchemaArray,
+    isValidField,
+} from 'scenes/pipeline/configUtils'
+import { PluginField } from 'scenes/plugins/edit/PluginField'
+
+import { pipelineAppLogic } from './pipelineAppLogic'
+
+export function PipelineAppConfiguration(): JSX.Element {
+    const { appType } = useValues(pipelineAppLogic)
+
+    if (appType === 'webhook') {
+        return (
+            <LemonWidget title="Configuration">
+                <WebhookAppConfiguration />
+            </LemonWidget>
+        )
+    }
+
+    return <p>Unsupported app type</p>
+}
+
+function WebhookAppConfiguration(): JSX.Element {
+    const { maybePlugin, maybePluginConfig, configuration } = useValues(pipelineAppLogic)
+    const { resetConfiguration, setConfigurationValues } = useActions(pipelineAppLogic)
+
+    const [invisibleFields, setInvisibleFields] = useState<string[]>([])
+    const [requiredFields, setRequiredFields] = useState<string[]>([])
+
+    const updateInvisibleAndRequiredFields = (): void => {
+        setInvisibleFields(
+            maybePlugin ? determineInvisibleFields((fieldName) => configuration[fieldName], maybePlugin) : []
+        )
+        setRequiredFields(
+            maybePlugin ? determineRequiredFields((fieldName) => configuration[fieldName], maybePlugin) : []
+        )
+    }
+
+    useEffect(() => {
+        if (maybePlugin && maybePluginConfig) {
+            setConfigurationValues({
+                ...(maybePluginConfig.config || defaultConfigForPlugin(maybePlugin)),
+                __enabled: maybePluginConfig.enabled,
+            })
+            // TODO: Do we need this: generateApiKeysIfNeeded(form)
```

Comment:
we can ignore this per `select is_global, * from posthog_plugin where config_schema #>> '{}' LIKE '%ogApi%'` it's not used by any global plugins and we're not going to add any new usages here.